rundeckapp/grails-app/controllers/rundeck/controllers/WorkflowController.groovy

```diff
@@ -101,8 +96,12 @@ class WorkflowController extends ControllerBase implements PluginListRequired {
         }
         AuthContext auth = frameworkService.getAuthContextForSubject(request.subject)
         def fprojects = frameworkService.projectNames(auth).findAll{it != params.project}
+        Map<String, Object> projectAndFrameworkValues = frameworkService.getProjectProperties(params.project)
+        def dynamicProperties = frameworkService.getDynamicProperties(params['newitemtype'], projectAndFrameworkValues)
```

Comment:
The dynamic properties should only be evaluated for "plugin" steps.  In addition there are two types of plugin steps (node, or workflow), so you need to check the `params['newitemnodestep`]=='true'` (for a new item) or `item.nodeStep` (for existing item).